mono/sgen/sgen-memory-governor.c

```diff
@@ -494,7 +502,10 @@ sgen_memgov_init (size_t max_heap, size_t soft_limit, gboolean debug_allowance,
 	sgen_register_fixed_internal_mem_type (INTERNAL_MEM_LOG_ENTRY, sizeof (SgenLogEntry));
 
 	if (max_heap == 0)
+	{
+		sgen_gc_info.total_available_memory_bytes = mono_determine_physical_ram_size ();
```

Comment:
It's surprising that `total_available_memory_bytes` is initialized with physical ram size, not `mono_determine_physical_ram_available_size`.  but i'll admit I didn't look at the implementation of those two functions in detail to tell what the difference is.

Reply:
Well, total_available_memory_bytes returns to total amount of physical memory in the machine.  mono_determine_physical_ram_available_size() gives the total amount, minus stuff currently allocated to processes. 

The meaning of the total_available_memory_bytes field is defined here: https://docs.microsoft.com/en-us/dotnet/api/system.gcmemoryinfo.totalavailablememorybytes?view=netcore-3.0#System_GCMemoryInfo_TotalAvailableMemoryBytes

I took this " Otherwise, the value of the property is the physical memory on the machine that was available for the garbage collector to use when the last garbage collection occurred." to mean the total physical memory on the machine. But I guess it could mean total - in use. However, it then seems weird that in the case of hard limits it is not defined as hard limit - used, doesn't it?